test/e2e/egress_test.go

```diff
@@ -256,80 +284,121 @@ func testEgressCRUD(t *testing.T, data *TestData) {
+				if egress.Status.EgressNode != "" {
+					exists, err := hasIP(data, egress.Status.EgressNode, egress.Spec.EgressIP)
+					require.NoError(t, err, "Failed to check if IP exists on Node")
+					assert.True(t, exists, "Didn't find desired IP on Node")
 				}
-				return true, nil
-			})
-			require.NoError(t, err, "Expected egressIP=%s nodeName in %s, got egressIP=%s nodeName=%s", tt.expectedEgressIP, tt.expectedNodes.List(), egress.Spec.EgressIP, egress.Status.EgressNode)
-			if egress.Status.EgressNode != "" {
-				exists, err := hasIP(data, egress.Status.EgressNode, egress.Spec.EgressIP)
-				require.NoError(t, err, "Failed to check if IP exists on Node")
-				assert.True(t, exists, "Didn't find desired IP on Node")
-			}
-
-			checkEIPStatus := func(expectedUsed int) {
-				pool, err = data.crdClient.CrdV1alpha2().ExternalIPPools().Get(context.TODO(), pool.Name, metav1.GetOptions{})
-				require.NoError(t, err, "Failed to get ExternalIPPool")
-				assert.Equal(t, expectedUsed, pool.Status.Usage.Used, "ExternalIPPool status usage used num not match")
-				assert.Equal(t, tt.expectedTotal, pool.Status.Usage.Total, "ExternalIPPool status usage total num not match")
-				t.Logf("ExternalIPPool %s status: %+v", pool.Name, pool.Status)
-			}
-			checkEIPStatus(1)
 
-			err = data.crdClient.CrdV1alpha2().Egresses().Delete(context.TODO(), egress.Name, metav1.DeleteOptions{})
-			require.NoError(t, err, "Failed to delete Egress")
-			if egress.Status.EgressNode != "" {
-				exists, err := hasIP(data, egress.Status.EgressNode, egress.Spec.EgressIP)
-				require.NoError(t, err, "Failed to check if IP exists on Node")
-				assert.False(t, exists, "Found stale IP on Node")
-			}
-			checkEIPStatus(0)
-		})
+				checkEIPStatus := func(expectedUsed int) {
+					pool, err = data.crdClient.CrdV1alpha2().ExternalIPPools().Get(context.TODO(), pool.Name, metav1.GetOptions{})
+					require.NoError(t, err, "Failed to get ExternalIPPool")
+					assert.Equal(t, expectedUsed, pool.Status.Usage.Used, "ExternalIPPool status usage used num not match")
+					assert.Equal(t, expectedTotal, pool.Status.Usage.Total, "ExternalIPPool status usage total num not match")
+					t.Logf("ExternalIPPool %s status: %+v", pool.Name, pool.Status)
+				}
+				checkEIPStatus(1)
+
+				err = data.crdClient.CrdV1alpha2().Egresses().Delete(context.TODO(), egress.Name, metav1.DeleteOptions{})
+				require.NoError(t, err, "Failed to delete Egress")
+				if egress.Status.EgressNode != "" {
+					var exists bool
+					err := wait.PollImmediate(200*time.Millisecond, 200*time.Second, func() (done bool, err error) {
+						exists, err = hasIP(data, egress.Status.EgressNode, egress.Spec.EgressIP)
+						if err != nil {
+							return false, err
+						}
+						return exists == false, nil
+					})
+					require.NoError(t, err, "Failed to check if IP exists on Node")
+					// assert.False(t, exists, "Found stale IP on Node")
```

Comment:
remove if not needed

Reply:
It seems that I comment this line by mistake, will fix it.